blockstore/src/main/java/brooklyn/location/blockstore/BlockDeviceOptions.java

```diff
@@ -27,17 +28,24 @@ public static BlockDeviceOptions fromMap(Map<String, ?> map) {
             }
         }
         if (map.containsKey("sizeInGb")) {
+            checkNotNull(map.get("sizeInGb"), "sizeInGb should not be null");
             if (map.get("sizeInGb") instanceof Double && Math.abs((Double) map.get("sizeInGb") - ((Double) map.get("sizeInGb")).intValue()) >= 0.01
                     || map.get("sizeInGb") instanceof Float && Math.abs((Float) map.get("sizeInGb") - ((Float) map.get("sizeInGb")).intValue()) >= 0.01) {
                 throw new UnsupportedOperationException("Trying to set block device with not allowed sizeInGb value "
                         + map.get("sizeInGb") + "; sizeInGb must have integer value.");
             } else if (map.get("sizeInGb") instanceof Double) {
-                result.sizeInGb = checkNotNull(((Double) map.get("sizeInGb")).intValue(), "sizeInGb");
+                result.sizeInGb = ((Double) map.get("sizeInGb")).intValue();
             } else if (map.get("sizeInGb") instanceof Float) {
-                result.sizeInGb = checkNotNull(((Float) map.get("sizeInGb")).intValue(), "sizeInGb");
+                result.sizeInGb = ((Float) map.get("sizeInGb")).intValue();
+            } else if (map.get("sizeInGb") instanceof Integer){
+                result.sizeInGb = (Integer)map.get("sizeInGb");
             } else {
-                result.sizeInGb = (int) checkNotNull(map.get("sizeInGb"), "sizeInGb");
+                result.sizeInGb = Integer.parseInt((String)map.get("sizeInGb"));
```

Comment:
This doesn't look right to me @bostko. You'd usually set numbers in the YAML as numbers (not wrapped in quotes). So it would be an Integer object in the map.
Still supporting a string is a good idea - consider using coerce instead.